test_lbaas.py

```diff
@@ -0,0 +1,625 @@
+                                                monitor_status):
+        load_balancer.refresh()
+        assert load_balancer.status == load_balancer_status
+        assert (load_balancer.pool_members[0]['monitor_status']
+                == monitor_status)
+
+    # Verify the health monitor reports the backend as up
+    retry_for(seconds=20).or_fail(
+        assert_load_balancer_and_monitor_status,
+        msg='Health monitor does not report "up" status after 20s',
+        load_balancer_status='running',
+        monitor_status='up',
+    )
+
+    # Shutdown the backend server and verify the health monitor goes down
+    backend.stop()
+    retry_for(seconds=20).or_fail(
+        assert_load_balancer_and_monitor_status,
+        msg='Health monitor does not report "down" status after 20s',
+        # As all pool members are down the LB is in status "error"
+        load_balancer_status='error',
+        monitor_status='down',
+    )
+
+    # Start the backend again and verfiy the health monitor goes up
+    # The test web server on the backend has to be started as well. It's not
+    # configured as a persistent systemd unit.
+    backend.start()
+    setup_lbaas_http_test_server(backend, ssl)
+    retry_for(seconds=20).or_fail(
+        assert_load_balancer_and_monitor_status,
+        msg='Health monitor does not report "up" status after 20s',
+        load_balancer_status='running',
+        monitor_status='up',
+    )
+
+
+@pytest.mark.parametrize('action', ('disable-enable', 'remove-add'))
+def test_pool_member_change(server, create_load_balancer_scenario,
+                            action):
+    """ A pool member can be removed and added back or disabled and reenabled
+    to a load balancer pool without any disruption to connections not
+    terminated on this pool member.
+
+    """
+
+    # This test uses a function scoped prober because we are going to start
+    # long running downloads which should not impact other tests.
+    prober = server
+
+    # Create a load balancer setup with 2 backends on a private network
+    load_balancer, listener, pool, backends, private_network = \
+        create_load_balancer_scenario(
+            num_backends=2,
+        )
+
+    # download an endless file to create "persistent" connections
+    backend_first = start_persistent_download(prober, load_balancer, backends,
+                                              'wget-first')
+
+    # second download to create a "persistent" connection to the other backend
+    backend_second = start_persistent_download(prober, load_balancer, backends,
+                                               'wget-second')
+
+    # Assert both backends got a request
+    assert set(backends) == {backend_first, backend_second}
+
+    # Get pool member serving the first download
+    member_first_name = \
+        f'{RESOURCE_NAME_PREFIX}-pool-member-{backend_first.name}'
+    member_first = next(filter(
```

Comment:
Why use a filter here? Why not just `next(x for x in load_balancer.pool_members if x['name'] == member_first_name)`

Reply:
If you think that's more readable. I don't care.

Reply:
Since `filter` is considered un-pythonic, and was almost dropped in Python 3, I would adjust that:
https://www.artima.com/weblogs/viewpost.jsp?thread=98196